### PR TITLE
889 e2e test sauron access

### PR DESCRIPTION
**Pull request by RyanRConaway**
## [Ticket Link](https://app.zenhub.com/workspaces/mdot-615b97c1a5fde400126174f8/issues/urbanos-public/internal/889)

Remove if not applicable

## Description

- Removed ids in resource creation bodies.
- Added additional error parsing in E2E test script:
- Added vim for quality of life in Sauron

## Reminders

- [ ] Did you up the relevant chart version numbers? (If appropriate)
  - [ ] If you up a chart version within urban-os, have you also upped the urban-os chart version itself?
  - [ ] If charts within the urban-os chart (andi, etc) have been updated, have you run `helm dependency update` in /charts/urban-os and commited the Chart.lock file?
- [ ] Does `helm template . -f values.yaml` pass? (Checks for default values provided in chart and catches other errors)
- [ ] Do you have git hooks installed? (See README.md to install)
- [ ] If global values were altered, are they included as chart default values?
  - [ ] Are they also specified in the urbanos chart values file?
- [ ] If references to external charts were added:
  - [ ] Was the github release action updated to `helm update {new_thing}` it's dependencies?
  - [ ] Was the deploy repo `-u` flag updated to `helm update {new_thing}` to ensure it's not left out of deployments?

### CODE REVIEW

File: charts/sauron/newman/e2e_test.postman_collection.json [hunk 1]
```diff
@@ -124,7 +131,9 @@
 					"listen": "prerequest",
 					"script": {
 						"exec": [
-							"setTimeout(() => {}, 2000)"
+							"setTimeout(() => {}, 2000)",
+							"",
+							"pm.expect(pm.collectionVariables.get(\"dataset_id\"), \"Could not obtain dataset_id from collection variables\").to.exist;"
```
Comment (RyanRConaway) [inline charts/sauron/newman/e2e_test.postman_collection.json:136]: Be sure the dataset_id is set before trying to create an ingestion

File: charts/sauron/newman/e2e_test.sh [hunk 2]
```diff
@@ -1,11 +1,32 @@
 #!/bin/bash
 
+OUTPUT_DIR=$1
+if [[ $OUTPUT_DIR == "" ]]; then
+  echo "Empty output location. Please provide an output location when running this script. EX: ./e2e_tests.sh /sauron"
+  exit 1
+fi
```
Comment (RyanRConaway) [inline charts/sauron/newman/e2e_test.sh:7]: Needed for permission issues in sauron. Can't write to /usr/local/bin, so we must specify the mounted pvc (/sauron)

File: charts/sauron/newman/e2e_test.sh [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+
+NORMALIZED_OUTPUT_DIR=$(dirname "$OUTPUT_DIR")/$(basename "$OUTPUT_DIR")
+
+OUTPUT_FILE="$NORMALIZED_OUTPUT_DIR/results.json"
+OUTPUT_ERROR_FILE="$NORMALIZED_OUTPUT_DIR/$(date +%s)_errors.json"
```
Comment (RyanRConaway) [inline charts/sauron/newman/e2e_test.sh:12]: timestamp the results file to allow for historical logs

File: charts/sauron/newman/e2e_test.sh [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
 newman run e2e_test.postman_collection.json --insecure --delay-request 2000 \
-    --reporters json --reporter-json-export results.json
+    --reporters json --reporter-json-export "$OUTPUT_FILE"
+
+FAILURES=$(jq .run.stats.assertions.failed "$OUTPUT_FILE")
+if [[ $FAILURES -gt 0 ]]; then
+  cat "$OUTPUT_FILE" | jq '.run.executions | map(.assertions) | flatten(2) | map(select(.error != null))' > $OUTPUT_ERROR_FILE
+
+  cat "$OUTPUT_ERROR_FILE"
+  echo "E2E TEST FAILURES: $FAILURES"
```
Comment (RyanRConaway) [inline charts/sauron/newman/e2e_test.sh:27]: Print the test results and the quantity

File: charts/sauron/newman/e2e_test.sh [hunk 5]
```diff
@@ -1,11 +1,32 @@
...
 echo "STARTING E2E TESTS"
 
 # setup
 newman run e2e_test.postman_collection.json --insecure --delay-request 2000 \
-    --reporters json --reporter-json-export results.json
+    --reporters json --reporter-json-export "$OUTPUT_FILE"
+
+FAILURES=$(jq .run.stats.assertions.failed "$OUTPUT_FILE")
+if [[ $FAILURES -gt 0 ]]; then
+  cat "$OUTPUT_FILE" | jq '.run.executions | map(.assertions) | flatten(2) | map(select(.error != null))' > $OUTPUT_ERROR_FILE
```
Comment (RyanRConaway) [inline charts/sauron/newman/e2e_test.sh:24]: jq magic to grab all assertion errors

File: charts/sauron/values.yaml [hunk 6]
```diff
@@ -5,7 +5,7 @@ global:
 
 config:
   cron: "*/10 * * * *"
-  image: "smartcitiesdata/sauron:0.0.5"
+  image: "smartcitiesdata/sauron:0.0.7"
```
Comment (ian-j-abbott-accenture) [inline charts/sauron/values.yaml:8]: doesn't matter at all, but any reason for skipping 0.0.6?

**Review by c-m-duncan — APPROVED** [review]
LGTM!

File: charts/sauron/newman/README.md [hunk 7]
```diff
@@ -15,6 +15,10 @@ Newman Tests run against production and goes from Andi Org creation through Disc
 
 ## To Run the suite
 - Enter Sauron (in environment or locally)
+  - Find the name of a previously run sauron cron pod - i.e:
+```shell
+oc debug sauron-cron-27900780--1-847ds
+```
```
Comment (bmitchinson) [inline charts/sauron/newman/README.md:21]: Will this execute the sauron deploy script as part of spinning up the pod? I suppose that's fine if so.

**Review by bmitchinson — APPROVED** [review]
lgtm

File: charts/sauron/newman/README.md (hunk 7 above)
Comment (RyanRConaway) [reply charts/sauron/newman/README.md:21]: It does not execute the entrypoint command

**Pull request merged**